src/validation.cpp

```diff
@@ -2467,12 +2468,23 @@ bool ActivateBestChain(CValidationState &state, const CChainParams& chainparams,
     // far from a guarantee. Things in the P2P/RPC will often end up calling
     // us in the middle of ProcessNewBlock - do not assume pblock is set
     // sanely for performance or correctness!
+    AssertLockNotHeld(cs_main);
 
     CBlockIndex *pindexMostWork = nullptr;
     CBlockIndex *pindexNewTip = nullptr;
     int nStopAtHeight = gArgs.GetArg("-stopatheight", DEFAULT_STOPATHEIGHT);
     do {
         boost::this_thread::interruption_point();
+
+        if (GetMainSignals().CallbacksPending() > 10) {
```

Comment:
Should this parameterized? Or at least a configurable constant?
Is there any advantage to tweaking it?

Reply:
During normal (read: non-IBD/reindex) usage, this should virtually never reach 10. During IBD/reindex, your memory is probably better speint in more dbcache than storing blocks you recently connected to eventually give to your wallet. I dont think more knobs is helpful here.

Reply:
https://github.com/bitcoin/bitcoin/pull/11824#discussion_r154814580

In commit "Block ActivateBestChain to empty validationinterface queue"

> During normal (read: non-IBD/reindex) usage, this should virtually never reach 10. During IBD/reindex, your memory is probably better speint in more dbcache than storing blocks you recently connected to eventually give to your wallet.

Please write this in a code comment. (How is anybody supposed to know this?)

Reply:
Done.

Reply:
It would be nice to introduce a named constant for this instead.